Main.java

```diff
@@ -0,0 +1,154 @@
+
+        MailMessage secondMessage = new MailMessage(
+                "Jonathan Nolan",
+                "Christopher Nolan",
+                "Брат, почему все так хвалят только тебя, когда практически все сценарии написал я. Так не честно!"
+        );
+
+        MailMessage thirdMessage = new MailMessage(
+                "Stephen Hawking",
+                "Christopher Nolan",
+                "Я так и не понял Интерстеллар."
+        );
+
+        List<MailMessage> messages = Arrays.asList(
+                firstMessage, secondMessage, thirdMessage
+        );
+
+        MailService<String> mailService = new MailService<>();
+
+        messages.stream().forEachOrdered(mailService);
+
+        Map<String, List<String>> mailBox = mailService.getMailBox();
+
+        assert mailBox.get("H.P. Lovecraft").equals(
+                Arrays.asList(
+                        "This \"The Shadow over Innsmouth\" story is real masterpiece, Howard!"
+                )
+        ) : "wrong mailService mailbox content (1)";
+
+        assert mailBox.get("Christopher Nolan").equals(
+                Arrays.asList(
+                        "Брат, почему все так хвалят только тебя, когда практически все сценарии написал я. Так не честно!",
+                        "Я так и не понял Интерстеллар."
+                )
+        ) : "wrong mailService mailbox content (2)";
+
+        assert mailBox.get(randomTo).equals(Collections.<String>emptyList()) : "wrong mailService mailbox content (3)";
+
+        Salary salary1 = new Salary("Facebook", "Mark Zuckerberg", 1);
+        Salary salary2 = new Salary("FC Barcelona", "Lionel Messi", Integer.MAX_VALUE);
+        Salary salary3 = new Salary(randomFrom, randomTo, randomSalary);
+
+        MailService<Integer> salaryService = new MailService<>();
+
+        Arrays.asList(salary1, salary2, salary3).forEach(salaryService);
+
+
+        Map<String, List<Integer>> salaries = salaryService.getMailBox();
+        assert salaries.get(salary1.getTo()).equals(Arrays.asList(1)) : "wrong salaries mailbox content (1)";
+        assert salaries.get(salary2.getTo()).equals(Arrays.asList(Integer.MAX_VALUE)) : "wrong salaries mailbox content (2)";
+        assert salaries.get(randomTo).equals(Arrays.asList(randomSalary)) : "wrong salaries mailbox content (3)";
+    }
+
+    public static Animal[] deserializeAnimalArray(byte[] data) {
+        try {
+            FileInputStream fis = new FileInputStream(String.valueOf(data));
+            ObjectInputStream ois = new ObjectInputStream(fis);
+            Animal[] animals = new Animal[ois.readInt()];
+            for (int i = 0; i < animals.length; i++) {
+                animals[i] = (Animal) ois.readObject();
+            }
+            ois.close();
+            fis.close();
+            return animals;
+        } catch (Exception e) {
+            throw new IllegalArgumentException("Illegal Argument Exception");

```

Comment:
print stacktrace

Reply:
just printStackTrace without throwing this useless IllegalArgumentException("Illegal Argument Exception")